test/contract.js

```diff
@@ -9,231 +9,247 @@ var StandAloneContract = require('../packages/web3-eth-contract');
+    "outputs": [{
+        "name": "d",
+        "type": "int"
     }]
 }, {
-    'name': 'Changed',
-    'type': 'event',
-    'inputs': [
-        {'name': 'from', 'type': 'address', 'indexed': true},
-        {'name': 'amount', 'type': 'uint256', 'indexed': true},
-        {'name': 't1', 'type': 'uint256', 'indexed': false},
-        {'name': 't2', 'type': 'uint256', 'indexed': false}
+    "name":"Changed",
+    "type":"event",
+    "inputs": [
+        {"name":"from","type":"address","indexed":true},
+        {"name":"amount","type":"uint256","indexed":true},
+        {"name":"t1","type":"uint256","indexed":false},
+        {"name":"t2","type":"uint256","indexed":false}
     ]
 }, {
-    'name': 'Unchanged',
-    'type': 'event',
-    'inputs': [
-        {'name': 'value', 'type': 'uint256', 'indexed': true},
-        {'name': 'addressFrom', 'type': 'address', 'indexed': true},
-        {'name': 't1', 'type': 'uint256', 'indexed': false}
-    ]
+        "name":"Unchanged",
+        "type":"event",
+        "inputs": [
+            {"name":"value","type":"uint256","indexed":true},
+            {"name":"addressFrom","type":"address","indexed":true},
+            {"name":"t1","type":"uint256","indexed":false}
+        ]
 }, {
-    'name': 'overloadedFunction',
-    'type': 'function',
-    'inputs': [
-        {'name': 'a', 'type': 'uint256'}
+    "name":"overloadedFunction",
+    "type":"function",
+    "inputs":[
+        {"name":"a","type":"uint256"}
     ],
-    'constant': true,
-    'outputs': [
-        {'name': '', 'type': 'uint256'}
+    "constant":true,
+    "outputs":[
+        {"name":"", "type":"uint256"}
     ],
-    'payable': false,
-    'stateMutability': 'view'
+    "payable":false,
+    "stateMutability":"view"
 }, {
-    'name': 'overloadedFunction',
-    'type': 'function',
-    'inputs': [],
-    'constant': true,
-    'outputs': [
-        {'name': '', 'type': 'uint256'}
+    "name":"overloadedFunction",
+    "type":"function",
+    "inputs":[],
+    "constant":true,
+    "outputs":[
+        {"name":"","type":"uint256"}
     ],
-    'payable': false,
-    'stateMutability': 'view'
-}];
+    "payable":false,
+    "stateMutability":"view"
+    }];
 
 var address = '0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe';
 var addressLowercase = '0x11f4d0a3c12e86b4b5f39b213f7e19d048276dae';
 var address2 = '0x5555567890123456789012345678901234567891';
-var account = {
-    address: '0xEB014f8c8B418Db6b45774c326A0E64C78914dC0',
-    privateKey: '0xbe6383dad004f233317e46ddb46ad31b16064d14447a95cc1d8c8d4bc61c3728'
-};
 
-/**
- * Factory method for standalone Contract objects.
- *
- * @method getStandAloneContractInstance
- *
- * @param {Array} abi
- * @param {String} address
- * @param {providers} provider
- * @param {Object} options
- *
- * @returns {Contract}
- */
-var getStandAloneContractInstance = function (abi, address, provider, options) {
+var getStandAloneContractInstance = function(abi, address, options, provider) {
```

Comment:
I've improved those two factory functions because line 195-213 is not required if you change the function parameters order. This means I would vote to keep those changes (``getStandAloneContractInstance`` & ``getEthContractInstance``).

Reply:
I think we disagree about this - the simplification you propose in #3190 is good, but it makes the file have a large number of changes that aren't strictly necessary to get the tests to pass. 

I would prefer to see the existing tests go in with an absolute minimum of difference so they are easily legible if someone looks at the PR later to figure what went wrong or what might have changed. 

With this diff, we will see that only the data type of subscriptions object and the question about eth.setProvider presented difficulties in getting the existing tests to pass.

Reply:
This is totally true. It will be way easier to follow the changes of PR #3190 this way 👌 (we can improve such stuff later)